src/hooks/useCodapState.tsx

```diff
@@ -174,8 +174,31 @@ export const useCodapState = () => {
     updateCollections();
   };
 
-  const handleAddAttribute = async (collection: ICollection) => {
-    const newAttributeName = `newAttr${collection.attrs.length + 1}`;
+  const handleAddAttribute = async (collection: ICollection, attrName: string) => {
+    const proposedName = attrName.length ? attrName : "newAttr";
+    let newAttributeName;
+    const attrNameAlreadyUsed = collection.attrs.find((attr) => attr.name === proposedName);
+    if (!attrNameAlreadyUsed) {
+      newAttributeName = proposedName;
+    } else {
+      const attrsWithSameName = collection.attrs.filter((attr) => attr.name.includes(proposedName));
```

Comment:
I think you want to use `.startsWith()` here instead of `.includes()` since you slice on the index into the string on the next line.